### DISCUSSION TITLE
Hono example

### DISCUSSION THREAD

**Discussion by itsdillon** [vikejs/vike#1239] [Discussion] [answered]
I'm having a tough time getting vike working with honojs. Any one have an example of getting it working?

**Comment by brillout** [comment]
> tough time getting vike working with honojs

Do you mind elaborating?

**Reply by itsdillon** [reply]
Ya for sure. I'm having issues specifically with vite dev server and using the middleware. When I try to pass the request I get `Argument of type 'HonoRequest<any, unknown>' is not assignable to parameter of type 'IncomingMessage'.`

**Reply by brillout** [reply]
In general:
 1. The issue is unrelated to Vike. (Vike's middleware should be trivial to add to your Hono server.) You can create a minimal reproduction of your issue without Vike by using https://github.com/bluwy/create-vite-extra with `ssr-vanilla`.
 1. What you're trying to do won't work until https://github.com/vitejs/vite/discussions/14288 is implemented.
 1. Thus, I'd approach it the other way around: how can (your) Hono middlewares be added to Vite's dev server?

I recommend reaching out to Hono &mdash; they seem quite responsive. CC me if you do. A Vite/Vike + Hono example would be very neat indeed.

**Reply by yusukebe** [reply]
Hi 👋

I think we can accomplish that with a [Hono node-server adapter](https://github.com/honojs/node-server) and you can refer to the [Vite dev-server](https://github.com/honojs/vite-plugins/tree/main/packages/dev-server). Anyway, it depends on what you want to achieve. If you've tried but haven't resolved the issue, please feel free to post a comment on [Hono's discussion](https://github.com/orgs/honojs/discussions).

**Reply by yusukebe** [reply]
BTW, I'm very curious about this issue #https://github.com/vitejs/vite/discussions/14288. If it comes to fruition, many problems will be resolved.

**Reply by brillout** [reply]
Hi @yusukebe, thanks for chiming in 👍 and thank you for your fantastic work on Hono.

FYI @magne4000 and myself were thinking of adding Hono to [Bati](https://batijs.github.io/). So an example would be a good first step; @dillonraphael let us know how creating a Vike+Hono example goes.

> very curious about this issue #vitejs/vite#14288

Yes, I'm super excited by it as well 👀 (friendly ping @sapphi-red).

**Reply by sapphi-red** [reply]
Glad that you are interested! It'll take some more time because I want to do a bit more POCs and the vite-node PR needs to be merged.

**Comment by brillout** [comment, marked as answer]
Let's consolidate the discussion over here: https://github.com/vikejs/vike/discussions/1391.
